libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1344,16 +1398,8 @@ MQTTStatus_t MQTT_GetConnectPacketSize( const MQTTConnectInfo_t * pConnectInfo,
          * the "Remaining Length" field plus 1 byte for the "Packet Type" field. */
         connectPacketSize += 1U + remainingLengthEncodedSize( connectPacketSize );
 
-        /* Check that the CONNECT packet is within the bounds of the MQTT spec. */
-        if( connectPacketSize > MQTT_PACKET_CONNECT_MAX_SIZE )
-        {
-            status = MQTTBadParameter;
-        }
```

Comment:
I think some comments about why an overflow check is not required here would help. What do you think?

Reply:
Yes thanks